### ISSUE TITLE
Existing sequences in production backscope that do not work in frontscope

### ISSUE THREAD

**Issue by gwhitney** [numberscope/frontscope#252]
Visiting numberscope.colorado.edu, the Virahanka-Fibonacci numbers A000045 or the pitch sequence of Beethoven's Für Elise A123456 cannot be used via Get OEIS Sequence with the Differences Visualizer: the former pops up a "Caching -1 to 999" error box and the latter seems reliably to produce a Status Code 500 error. Yet visiting `numberscope.colorado.edu/api/get_oeis_values/A000045/1000` or similarly for A123456 produces a perfectly valid JSON response that includes correct values. So the problem appears to be on the frontend.

On the other hand, both primes A000040 and "old style primes with one" A008578 work both with the Differences visualizer and the api.

The only visible difference Kate and I could see was that the former two do not have their correct names in the backend database, whereas the latter two do.

This leads to two wild hypotheses, neither of which may have any bearing on reality:

1) The parentheses in the "placeholder" name that gets returned when the database has no name in it are causing the front end to barf, or

2) The bad sequences are not actually in the backscope  database, and it's just grabbing them on the fly from OEIS every time, and that takes just enough longer than returning a sequence from the database does that it's causing the frontend to barf.

Anyhow, sequences like the first two, especially because they tend to be common, popular series, are a big, embarrassing, blocking blot on our escutcheon at the moment, so this issue deserves top priority.

**Comment by katestange** [comment]
Thank you for teaching me a new word!

**Comment by gwhitney** [comment]
I don't know of any other use of "escutcheon" in the past century other than that phrase, and I am sure it's pretty dated even so (note I gave myself a century there!) ;-)

**Comment by gwhitney** [comment]
Update 2 weeks later: Oddly A000045 is now working, both on numberscope.colorado.edu and Kate & Glen's local dev frontends, but A123456 is still causing cross-origin errors. Strange.

**Comment by liammulh** [comment]
As of this writing:

- numberscope.colorado.edu's backscope is on https://github.com/numberscope/backscope/commit/91c4a7d07e0ab00331f0619a163bcef8fb2215ce.
- numberscope.colorado.edu's frontscope is on https://github.com/numberscope/frontscope/commit/de1ded572b722db07d5b1e9b70a02bc84648174a.

**Comment by liammulh** [comment]
Add the below patch to your local copy if you'd like to see errors in the console.

<details>

<summary>Patch</summary>

```
diff --git a/src/main.ts b/src/main.ts
index 2fa7f45..fb4c196 100644
--- a/src/main.ts
+++ b/src/main.ts
@@ -6,6 +6,7 @@ import {alertMessage} from './shared/alertMessage'
 const app = createApp(App)
 app.config.errorHandler = e => {
     window.alert(alertMessage(e))
+    console.error(e)
 }
 
 app.use(router)
```

</details>

When I added the above patch, on macOS 12.4 + Chrome 111.0.5563.64, I saw:

```
Access to XMLHttpRequest at 'https://numberscope.colorado.edu/api/get_oeis_factors/A123456/1000' from origin 'http://127.0.0.1:5173' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by gwhitney** [comment]
Does the call to window.alert() block until the person using the browser has responded to the dialog? In other words, will the console error not occur until the error dialog is dismissed? If so, should the order in the code be changed? What's your thought about putting this patch into branch main? Seems useful to me and I think it's OK to write to the console in an error condition...

**Comment by liammulh** [comment]
> I don't have an issue with logging errors in production.

... as long as we are using `console.error` rather than `console.log`.

**Comment by liammulh** [comment]
I added a print statement to view the response from the server:

<details>

<summary>Patch</summary>

```
diff --git a/src/sequences/OEIS.ts b/src/sequences/OEIS.ts
index 76f4323..e84515f 100644
--- a/src/sequences/OEIS.ts
+++ b/src/sequences/OEIS.ts
@@ -1,6 +1,6 @@
 import type {ValidationStatus} from '../shared/ValidationStatus'
 import {modulo} from '../shared/math'
-import {SequenceExportModule, SequenceExportKind} from './SequenceInterface'
+import {SequenceExportKind, SequenceExportModule} from './SequenceInterface'
 import {Cached} from './Cached'
 import {alertMessage} from '../shared/alertMessage'
 
@@ -55,6 +55,8 @@ export default class OEIS extends Cached {
                 urlPrefix
                 + `get_oeis_values/${this.oeisId}/${this.cacheBlock}`
             const response = await axios.get(backendUrl)
+            // eslint-disable-next-line max-len
+            console.log( `----------> response = ${JSON.stringify( response, null, 4 )}` );
             this.first = Infinity
             this.last = -Infinity
             for (const k in response.data.values) {
@@ -78,8 +80,8 @@ export default class OEIS extends Cached {
             } else {
                 // OK, now get the factors
                 const factorUrl =
-                    urlPrefix
-                    + `get_oeis_factors/${this.oeisId}/${this.cacheBlock}`
+                    urlPrefix +
+                    `get_oeis_factors/${this.oeisId}/${this.cacheBlock}`
                 const factorResponse = await axios.get(factorUrl)
                 for (const k in factorResponse.data.factors) {
                     const index = Number(k)
@@ -119,16 +121,16 @@ export default class OEIS extends Cached {
         const status = super.checkParameters()
 
         if (
-            this.params.oeisId.value.length !== 7
-            || (this.params.oeisId.value[0] !== 'A'
-                && this.params.oeisId.value[0] !== 'a')
+            this.params.oeisId.value.length !== 7 ||
+            (this.params.oeisId.value[0] !== 'A' &&
+                this.params.oeisId.value[0] !== 'a')
         ) {
             status.errors.push('OEIS IDs are of form Annnnnn')
         }
         if (typeof this.params.cacheBlock.value === 'number') {
             if (
-                this.params.cacheBlock.value < 0
-                || !Number.isInteger(this.params.cacheBlock.value)
+                this.params.cacheBlock.value < 0 ||
+                !Number.isInteger(this.params.cacheBlock.value)
             ) {
                 status.errors.push(
                     'Number of elements must be a positive integer.'
```

</details>

<details>

<summary>Response</summary>

```
----------> response = {
    "data": {
        "id": "A123456",
        "name": "A123456 [name not yet loaded]",
        "values": {
            "1": "-20",
            "2": "56",
            "3": "55",
            "4": "56",
            "5": "55",
            "6": "56",
            "7": "51",
            "8": "54",
            "9": "52",
            "10": "49",
            "11": "32",
            "12": "37",
            "13": "40",
            "14": "44",
            "15": "49",
            "16": "51",
            "17": "32",
            "18": "36",
            "19": "44",
            "20": "48",
            "21": "51",
            "22": "52",
            "23": "32",
            "24": "37",
            "25": "44",
            "26": "56",
            "27": "55",
            "28": "56",
            "29": "55",
            "30": "56",
            "31": "51",
            "32": "54",
            "33": "52",
            "34": "49",
            "35": "32",
            "36": "37",
            "37": "40",
            "38": "44",
            "39": "49",
            "40": "51",
            "41": "32",
            "42": "36",
            "43": "44",
            "44": "52",
            "45": "51",
            "46": "49",
            "47": "32",
            "48": "37",
            "49": "-20",
            "50": "56",
            "51": "55",
            "52": "56",
            "53": "55",
            "54": "56",
            "55": "51",
            "56": "54",
            "57": "52",
            "58": "49",
            "59": "32",
            "60": "37",
            "61": "40",
            "62": "44",
            "63": "49",
            "64": "51",
            "65": "32",
            "66": "36",
            "67": "44",
            "68": "48",
            "69": "51",
            "70": "52",
            "71": "32",
            "72": "37",
            "73": "44",
            "74": "56",
            "75": "55",
            "76": "56",
            "77": "55",
            "78": "56",
            "79": "51",
            "80": "54",
            "81": "52",
            "82": "49",
            "83": "32",
            "84": "37",
            "85": "40",
            "86": "44",
            "87": "49",
            "88": "51",
            "89": "32",
            "90": "36",
            "91": "44",
            "92": "52",
            "93": "51",
            "94": "49",
            "95": "32",
            "96": "37",
            "97": "51",
            "98": "52",
            "99": "54",
            "100": "56",
            "101": "35",
            "102": "40",
            "103": "47",
            "104": "57",
            "105": "56",
            "106": "54",
            "107": "35",
            "108": "39",
            "109": "45",
            "110": "56",
            "111": "54",
            "112": "52",
            "113": "32",
            "114": "37",
            "115": "44",
            "116": "54",
            "117": "52",
            "118": "51",
            "119": "32",
            "120": "44",
            "121": "44",
            "122": "56",
            "123": "44",
            "124": "56",
            "125": "56",
            "126": "68",
            "127": "55",
            "128": "56",
            "129": "55",
            "130": "56",
            "131": "55",
            "132": "56",
            "133": "55",
            "134": "56",
            "135": "55",
            "136": "56",
            "137": "55",
            "138": "56",
            "139": "51",
            "140": "54",
            "141": "52",
            "142": "49",
            "143": "32",
            "144": "37",
            "145": "40",
            "146": "44",
            "147": "49",
            "148": "51",
            "149": "32",
            "150": "36",
            "151": "44",
            "152": "48",
            "153": "51",
            "154": "52",
            "155": "32",
            "156": "37",
            "157": "44",
            "158": "56",
            "159": "55",
            "160": "56",
            "161": "55",
            "162": "56",
            "163": "51",
            "164": "54",
            "165": "52",
            "166": "49",
            "167": "32",
            "168": "37",
            "169": "40",
            "170": "44",
            "171": "49",
            "172": "51",
            "173": "32",
            "174": "36",
            "175": "44",
            "176": "52",
            "177": "51",
            "178": "49",
            "179": "32",
            "180": "37",
            "181": "52",
            "182": "44",
            "183": "52",
            "184": "45",
            "185": "52",
            "186": "47",
            "187": "52",
            "188": "33",
            "189": "37",
            "190": "-20",
            "191": "40",
            "192": "-20",
            "193": "37",
            "194": "-20",
            "195": "57",
            "196": "40",
            "197": "37",
            "198": "56",
            "199": "56",
            "200": "33",
            "201": "38",
            "202": "-20",
            "203": "54",
            "204": "42",
            "205": "38",
            "206": "-20",
            "207": "62",
            "208": "42",
            "209": "38",
            "210": "61",
            "211": "61",
            "212": "35",
            "213": "59",
            "214": "44",
            "215": "57",
            "216": "35",
            "217": "56",
            "218": "44",
            "219": "54",
            "220": "35",
            "221": "52",
            "222": "44",
            "223": "50",
            "224": "33",
            "225": "37",
            "226": "-20",
            "227": "49",
            "228": "40",
            "229": "37",
            "230": "50",
            "231": "49",
            "232": "47",
            "233": "49",
            "234": "50",
            "235": "52",
            "236": "33",
            "237": "37",
            "238": "-20",
            "239": "40",
            "240": "-20",
            "241": "37",
            "242": "-20",
            "243": "54",
            "244": "40",
            "245": "55",
            "246": "37",
            "247": "56",
            "248": "32",
            "249": "37",
            "250": "-20",
            "251": "40",
            "252": "-20",
            "253": "56",
            "254": "37",
            "255": "57",
            "256": "42",
            "257": "49",
            "258": "37",
            "259": "52",
            "260": "35",
            "261": "44",
            "262": "-20",
            "263": "54",
            "264": "52",
            "265": "51",
            "266": "52",
            "267": "54",
            "268": "45",
            "269": "35",
            "270": "51",
            "271": "40",
            "272": "59",
            "273": "47",
            "274": "59",
            "275": "49",
            "276": "59",
            "277": "51",
            "278": "59",
            "279": "52",
            "280": "59",
            "281": "54",
            "282": "59",
            "283": "56",
            "284": "59",
            "285": "64",
            "286": "63",
            "287": "33",
            "288": "59",
            "289": "57",
            "290": "56",
            "291": "35",
            "292": "59",
            "293": "57",
            "294": "54",
            "295": "40",
            "296": "59",
            "297": "47",
            "298": "59",
            "299": "49",
            "300": "59",
            "301": "51",
            "302": "59",
            "303": "52",
            "304": "59",
            "305": "54",
            "306": "59",
            "307": "56",
            "308": "59",
            "309": "64",
            "310": "63",
            "311": "33",
            "312": "59",
            "313": "57",
            "314": "56",
            "315": "35",
            "316": "59",
            "317": "57",
            "318": "54",
            "319": "36",
            "320": "57",
            "321": "56",
            "322": "55",
            "323": "56",
            "324": "51",
            "325": "56",
            "326": "55",
            "327": "56",
            "328": "51",
            "329": "56",
            "330": "55",
            "331": "56",
            "332": "-20",
            "333": "-20",
            "334": "51",
            "335": "56",
            "336": "55",
            "337": "56",
            "338": "-20",
            "339": "-20",
            "340": "51",
            "341": "56",
            "342": "55",
            "343": "56",
            "344": "55",
            "345": "56",
            "346": "55",
            "347": "56",
            "348": "55",
            "349": "56",
            "350": "55",
            "351": "56",
            "352": "55",
            "353": "56",
            "354": "55",
            "355": "56",
            "356": "55",
            "357": "56",
            "358": "51",
            "359": "54",
            "360": "52",
            "361": "49",
            "362": "32",
            "363": "37",
            "364": "40",
            "365": "44",
            "366": "49",
            "367": "51",
            "368": "32",
            "369": "36",
            "370": "44",
            "371": "48",
            "372": "51",
            "373": "52",
            "374": "32",
            "375": "37",
            "376": "44",
            "377": "56",
            "378": "55",
            "379": "56",
            "380": "55",
            "381": "56",
            "382": "51",
            "383": "54",
            "384": "52",
            "385": "49",
            "386": "32",
            "387": "37",
            "388": "40",
            "389": "44",
            "390": "49",
            "391": "51",
            "392": "32",
            "393": "36",
            "394": "44",
            "395": "52",
            "396": "51",
            "397": "49",
            "398": "32",
            "399": "37",
            "400": "51",
            "401": "52",
            "402": "54",
            "403": "56",
            "404": "35",
            "405": "40",
            "406": "47",
            "407": "57",
            "408": "56",
            "409": "54",
            "410": "35",
            "411": "39",
            "412": "45",
            "413": "56",
            "414": "54",
            "415": "52",
            "416": "32",
            "417": "37",
            "418": "44",
            "419": "54",
            "420": "52",
            "421": "51",
            "422": "32",
            "423": "44",
            "424": "44",
            "425": "56",
            "426": "44",
            "427": "56",
            "428": "56",
            "429": "68",
            "430": "55",
            "431": "56",
            "432": "55",
            "433": "56",
            "434": "55",
            "435": "56",
            "436": "55",
            "437": "56",
            "438": "55",
            "439": "56",
            "440": "55",
            "441": "56",
            "442": "51",
            "443": "54",
            "444": "52",
            "445": "49",
            "446": "32",
            "447": "37",
            "448": "40",
            "449": "44",
            "450": "49",
            "451": "51",
            "452": "32",
            "453": "36",
            "454": "44",
            "455": "48",
            "456": "51",
            "457": "52",
            "458": "32",
            "459": "37",
            "460": "44",
            "461": "56",
            "462": "55",
            "463": "56",
            "464": "55",
            "465": "56",
            "466": "51",
            "467": "54",
            "468": "52",
            "469": "49",
            "470": "32",
            "471": "37",
            "472": "40",
            "473": "44",
            "474": "49",
            "475": "51",
            "476": "32",
            "477": "36",
            "478": "44",
            "479": "52",
            "480": "51",
            "481": "49",
            "482": "25",
            "483": "25",
            "484": "25",
            "485": "25",
            "486": "25",
            "487": "53",
            "488": "25",
            "489": "25",
            "490": "25",
            "491": "25",
            "492": "25",
            "493": "54",
            "494": "25",
            "495": "25",
            "496": "25",
            "497": "56",
            "498": "57",
            "499": "48",
            "500": "25",
            "501": "25",
            "502": "25",
            "503": "57",
            "504": "25",
            "505": "56",
            "506": "25",
            "507": "25",
            "508": "25",
            "509": "25",
            "510": "25",
            "511": "54",
            "512": "18",
            "513": "18",
            "514": "18",
            "515": "52",
            "516": "51",
            "517": "49",
            "518": "19",
            "519": "19",
            "520": "19",
            "521": "49",
            "522": "19",
            "523": "49",
            "524": "20",
            "525": "52",
            "526": "20",
            "527": "51",
            "528": "24",
            "529": "49",
            "530": "25",
            "531": "25",
            "532": "25",
            "533": "25",
            "534": "25",
            "535": "53",
            "536": "25",
            "537": "25",
            "538": "25",
            "539": "25",
            "540": "25",
            "541": "54",
            "542": "25",
            "543": "25",
            "544": "25",
            "545": "56",
            "546": "57",
            "547": "57",
            "548": "25",
            "549": "25",
            "550": "25",
            "551": "57",
            "552": "25",
            "553": "57",
            "554": "26",
            "555": "26",
            "556": "26",
            "557": "26",
            "558": "26",
            "559": "55",
            "560": "26",
            "561": "26",
            "562": "26",
            "563": "54",
            "564": "52",
            "565": "50",
            "566": "26",
            "567": "26",
            "568": "26",
            "569": "49",
            "570": "26",
            "571": "48",
            "572": "27",
            "573": "27",
            "574": "27",
            "575": "48",
            "576": "27",
            "577": "49",
            "578": "28",
            "579": "-20",
            "580": "-20",
            "581": "-20",
            "582": "-20",
            "583": "48",
            "584": "32",
            "585": "-20",
            "586": "-20",
            "587": "-20",
            "588": "-20",
            "589": "25",
            "590": "40",
            "591": "44",
            "592": "49",
            "593": "52",
            "594": "56",
            "595": "54",
            "596": "52",
            "597": "51",
            "598": "37",
            "599": "52",
            "600": "56",
            "601": "61",
            "602": "64",
            "603": "68",
            "604": "66",
            "605": "64",
            "606": "63",
            "607": "37",
            "608": "64",
            "609": "68",
            "610": "73",
            "611": "76",
            "612": "80",
            "613": "78",
            "614": "76",
            "615": "75",
            "616": "74",
            "617": "73",
            "618": "72",
            "619": "71",
            "620": "70",
            "621": "69",
            "622": "68",
            "623": "67",
            "624": "66",
            "625": "65",
            "626": "64",
            "627": "63",
            "628": "62",
            "629": "61",
            "630": "60",
            "631": "59",
            "632": "58",
            "633": "57",
            "634": "56",
            "635": "55",
            "636": "56",
            "637": "51",
            "638": "54",
            "639": "52",
            "640": "49",
            "641": "32",
            "642": "37",
            "643": "40",
            "644": "44",
            "645": "49",
            "646": "51",
            "647": "32",
            "648": "36",
            "649": "44",
            "650": "48",
            "651": "51",
            "652": "52",
            "653": "32",
            "654": "37",
            "655": "44",
            "656": "56",
            "657": "55",
            "658": "56",
            "659": "55",
            "660": "56",
            "661": "51",
            "662": "54",
            "663": "52",
            "664": "49",
            "665": "32",
            "666": "37",
            "667": "40",
            "668": "44",
            "669": "49",
            "670": "51",
            "671": "32",
            "672": "36",
            "673": "44",
            "674": "52",
            "675": "51",
            "676": "49",
            "677": "32",
            "678": "37",
            "679": "51",
            "680": "52",
            "681": "54",
            "682": "56",
            "683": "35",
            "684": "40",
            "685": "47",
            "686": "57",
            "687": "56",
            "688": "54",
            "689": "35",
            "690": "39",
            "691": "45",
            "692": "56",
            "693": "54",
            "694": "52",
            "695": "32",
            "696": "37",
            "697": "44",
            "698": "54",
            "699": "52",
            "700": "51",
            "701": "32",
            "702": "44",
            "703": "44",
            "704": "56",
            "705": "44",
            "706": "56",
            "707": "56",
            "708": "68",
            "709": "55",
            "710": "56",
            "711": "55",
            "712": "56",
            "713": "55",
            "714": "56",
            "715": "55",
            "716": "56",
            "717": "55",
            "718": "56",
            "719": "55",
            "720": "56",
            "721": "51",
            "722": "54",
            "723": "52",
            "724": "49",
            "725": "32",
            "726": "37",
            "727": "40",
            "728": "44",
            "729": "49",
            "730": "51",
            "731": "32",
            "732": "36",
            "733": "44",
            "734": "48",
            "735": "51",
            "736": "52",
            "737": "32",
            "738": "37",
            "739": "44",
            "740": "56",
            "741": "55",
            "742": "56",
            "743": "55",
            "744": "56",
            "745": "51",
            "746": "54",
            "747": "52",
            "748": "49",
            "749": "32",
            "750": "37",
            "751": "40",
            "752": "44",
            "753": "49",
            "754": "51",
            "755": "32",
            "756": "36",
            "757": "42",
            "758": "52",
            "759": "51",
            "760": "49",
            "761": "25"
        }
    },
    "status": 200,
    "statusText": "OK",
    "headers": {
        "content-type": "application/json"
    },
    "config": {
        "transitional": {
            "silentJSONParsing": true,
            "forcedJSONParsing": true,
            "clarifyTimeoutError": false
        },
        "transformRequest": [
            null
        ],
        "transformResponse": [
            null
        ],
        "timeout": 0,
        "xsrfCookieName": "XSRF-TOKEN",
        "xsrfHeaderName": "X-XSRF-TOKEN",
        "maxContentLength": -1,
        "maxBodyLength": -1,
        "env": {
            "FormData": null
        },
        "headers": {
            "Accept": "application/json, text/plain, */*"
        },
        "method": "get",
        "url": "https://numberscope.colorado.edu/api/get_oeis_values/A123456/1000"
    },
    "request": {}
}
```

</details>

The data object contains the sequence, and the status of the response is 200 for the URL https://numberscope.colorado.edu/api/get_oeis_values/A123456/1000.

**Comment by liammulh** [comment]
To be clear, I am still seeing the errors I mentioned in https://github.com/numberscope/frontscope/issues/252#issuecomment-1476729823, despite seeing the sequence in the response.

**Comment by gwhitney** [comment]
Ah, well that's some progress finally. Since you seem to have undertaken debugging this, I will refrain from embarking on anything before tomorrow's meeting. Thanks!

**Comment by liammulh** [comment]
I get "500 Internal Server Error" when I try to go to `https://numberscope.colorado.edu/api/get_oeis_factors/A123456/1000`.

**Comment by gwhitney** [comment]
Problem with the Pari/cypari installation? Just a wild guess based on the fact there has been issues with that in the past...

**Comment by liammulh** [comment]
When I try to go to `https://numberscope.colorado.edu/api/get_oeis_factors/A123456/1000`, I see these logs on the production server:

<details>

<summary>Logs</summary>

```
Mar 20 15:22:14 numberscope bash[3869558]: UnicodeEncodeError: 'ascii' codec can't encode character '\xfc' in position 42: ordinal not in range(128)
Mar 20 15:22:14 numberscope bash[3869558]: Exception on /api/get_oeis_factors/A123456/1000 [GET]
Mar 20 15:22:14 numberscope bash[3869558]: Traceback (most recent call last):
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
Mar 20 15:22:14 numberscope bash[3869558]:     response = self.full_dispatch_request()
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
Mar 20 15:22:14 numberscope bash[3869558]:     rv = self.handle_user_exception(e)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
Mar 20 15:22:14 numberscope bash[3869558]:     reraise(exc_type, exc_value, tb)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
Mar 20 15:22:14 numberscope bash[3869558]:     raise value
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
Mar 20 15:22:14 numberscope bash[3869558]:     rv = self.dispatch_request()
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
Mar 20 15:22:14 numberscope bash[3869558]:     return self.view_functions[rule.endpoint](**req.view_args)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/flaskr/nscope/views.py", line 258, in get_oeis_factors
Mar 20 15:22:14 numberscope bash[3869558]:     seq = find_oeis_sequence(valid_oeis_id, 'full_nofactor') # we're about to do it...
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/flaskr/nscope/views.py", line 131, in find_oeis_sequence
Mar 20 15:22:14 numberscope bash[3869558]:     return save_oeis_sequence(seq, detail)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/flaskr/nscope/views.py", line 70, in save_oeis_sequence
Mar 20 15:22:14 numberscope bash[3869558]:     db.session.commit()
Mar 20 15:22:14 numberscope bash[3869558]:   File "<string>", line 2, in commit
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1451, in commit
Mar 20 15:22:14 numberscope bash[3869558]:     self._transaction.commit(_to_root=self.future)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 829, in commit
Mar 20 15:22:14 numberscope bash[3869558]:     self._prepare_impl()
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 808, in _prepare_impl
Mar 20 15:22:14 numberscope bash[3869558]:     self.session.flush()
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3383, in flush
Mar 20 15:22:14 numberscope bash[3869558]:     self._flush(objects)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3523, in _flush
Mar 20 15:22:14 numberscope bash[3869558]:     transaction.rollback(_capture_exception=True)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
Mar 20 15:22:14 numberscope bash[3869558]:     compat.raise_(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
Mar 20 15:22:14 numberscope bash[3869558]:     raise exception
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3483, in _flush
Mar 20 15:22:14 numberscope bash[3869558]:     flush_context.execute()
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/unitofwork.py", line 456, in execute
Mar 20 15:22:14 numberscope bash[3869558]:     rec.execute(self)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/unitofwork.py", line 630, in execute
Mar 20 15:22:14 numberscope bash[3869558]:     util.preloaded.orm_persistence.save_obj(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
Mar 20 15:22:14 numberscope bash[3869558]:     _emit_insert_statements(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/orm/persistence.py", line 1097, in _emit_insert_statements
Mar 20 15:22:14 numberscope bash[3869558]:     c = connection._execute_20(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
Mar 20 15:22:14 numberscope bash[3869558]:     return meth(self, args_10style, kwargs_10style, execution_options)
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 333, in _execute_on_connection
Mar 20 15:22:14 numberscope bash[3869558]:     return connection._execute_clauseelement(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
Mar 20 15:22:14 numberscope bash[3869558]:     ret = self._execute_context(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
Mar 20 15:22:14 numberscope bash[3869558]:     self._handle_dbapi_exception(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2128, in _handle_dbapi_exception
Mar 20 15:22:14 numberscope bash[3869558]:     util.raise_(exc_info[1], with_traceback=exc_info[2])
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
Mar 20 15:22:14 numberscope bash[3869558]:     raise exception
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
Mar 20 15:22:14 numberscope bash[3869558]:     self.dialect.do_execute(
Mar 20 15:22:14 numberscope bash[3869558]:   File "/home/scope/repos/backscope/.venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
Mar 20 15:22:14 numberscope bash[3869558]:     cursor.execute(statement, parameters)
Mar 20 15:22:14 numberscope bash[3869558]: UnicodeEncodeError: 'ascii' codec can't encode character '\xfc' in position 42: ordinal not in range(128)
```

</details>

**Comment by liammulh** [comment]
The character being referenced is the u with an umlaut, I think. See https://www.codetable.net/hex/fc.

**Comment by gwhitney** [comment]
Oh, sequence A123456 is the tone sequence for Für Elise by Beethoven. Perhaps the non-ascii character in the description of the sequence could be the culprit? (The backend should really handle arbitrary Unicode descriptions.)

**Comment by gwhitney** [comment]
Feel free to try, and/or we can discuss tomorrow at the meeting.

**Comment by gwhitney** [comment]
OK, well the proposed patch is at least by itself not the solution. If it is used, then the name of A123456 is stored as "\x4c75647769672076616e2042656574686f76656e2c204261676174656c6c65204e6f2e2032352c202246c3bc7220456c697365222e" and reported as such. While that may be better than an error, it's not too useful. It is also the case that Postgres has no trouble storing utf-8 strings (I just manually changed the name in the database to the string with the umlaut and then it reports it fine). So it seems like we need a scheme that properly gets the utf-8 string into the database; then it should work just fine thereafter.

**Comment by gwhitney** [comment]
Depends on #260 

likely improved by https://github.com/numberscope/backscope/pull/67

When those two are both merged, there may be nothing left to resolve here. (fingers crossed)

**Comment by gwhitney** [comment]
This issue should be revisited once https://github.com/numberscope/backscope/pull/127 is resolved; hopefully that will lay this to rest, finally.

**Comment by gwhitney** [comment]
Closing as duplicate of #407.
